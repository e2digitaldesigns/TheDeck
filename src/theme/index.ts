export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
