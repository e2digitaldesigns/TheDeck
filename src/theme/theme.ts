export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
