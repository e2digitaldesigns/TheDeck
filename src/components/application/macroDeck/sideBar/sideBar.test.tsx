import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import SideBar from "./sideBar";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../theme/theme";

const setup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <SideBar />
    </ThemeProvider>
  );
};

describe("<SideBar/>", () => {
  it("Should render correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
