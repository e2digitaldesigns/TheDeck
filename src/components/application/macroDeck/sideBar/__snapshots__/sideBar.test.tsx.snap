// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar/> Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL loYPBS"
        data-testid="sidebar_header__component"
      >
        <div
          class="sc-gsDKAQ iKnNXR"
        >
          <div
            class="sc-dkPtRN gMLSJG"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
              />
            </svg>
          </div>
          <div>
            PROFILES
             (
            <span
              data-testid="sidebar-header__profile-count"
            >
              0
            </span>
            )
          </div>
          <div
            class="sc-dkPtRN sc-hKwDye gMLSJG kfsJTy"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              data-testid="sidebar-header__new-profile-button"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
          </div>
        </div>
        <div
          class="profile-search-wrapper"
        />
      </div>
      <div
        class="sc-jrQzAO fKtDVo"
      >
        <div
          class="sc-pVTFL sc-kDTinF ggqylo icXZMB"
        />
      </div>
      <div
        class="sc-iqseJM gZAsSF"
        data-testid="sidebar_style_header__component"
      >
        <div
          class="sc-crHmcD bkPcYp"
        >
          <div
            class="sc-egiyK eSsMSI"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
              />
            </svg>
          </div>
          <div>
            STYLES
             (
            <span
              data-testid="sidebar_style_header__style-count"
            >
              0
            </span>
            )
          </div>
          <div
            class="sc-egiyK sc-bqiRlB eSsMSI kmdMfI"
          />
        </div>
        <div
          class="sc-ksdxgE bSMgFf"
          data-testid="sidebar_style_header__drop-zone"
        >
          Drop Pad Here! 
          <br />
          to save button style...
        </div>
      </div>
      <div
        class="sc-jrQzAO hsAljN"
        height="12em"
      >
        <div
          class="sc-pVTFL sc-kDTinF ggqylo icXZMB"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL loYPBS"
      data-testid="sidebar_header__component"
    >
      <div
        class="sc-gsDKAQ iKnNXR"
      >
        <div
          class="sc-dkPtRN gMLSJG"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
            />
          </svg>
        </div>
        <div>
          PROFILES
           (
          <span
            data-testid="sidebar-header__profile-count"
          >
            0
          </span>
          )
        </div>
        <div
          class="sc-dkPtRN sc-hKwDye gMLSJG kfsJTy"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            data-testid="sidebar-header__new-profile-button"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </div>
      </div>
      <div
        class="profile-search-wrapper"
      />
    </div>
    <div
      class="sc-jrQzAO fKtDVo"
    >
      <div
        class="sc-pVTFL sc-kDTinF ggqylo icXZMB"
      />
    </div>
    <div
      class="sc-iqseJM gZAsSF"
      data-testid="sidebar_style_header__component"
    >
      <div
        class="sc-crHmcD bkPcYp"
      >
        <div
          class="sc-egiyK eSsMSI"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
            />
          </svg>
        </div>
        <div>
          STYLES
           (
          <span
            data-testid="sidebar_style_header__style-count"
          >
            0
          </span>
          )
        </div>
        <div
          class="sc-egiyK sc-bqiRlB eSsMSI kmdMfI"
        />
      </div>
      <div
        class="sc-ksdxgE bSMgFf"
        data-testid="sidebar_style_header__drop-zone"
      >
        Drop Pad Here! 
        <br />
        to save button style...
      </div>
    </div>
    <div
      class="sc-jrQzAO hsAljN"
      height="12em"
    >
      <div
        class="sc-pVTFL sc-kDTinF ggqylo icXZMB"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
